### DISCUSSION TITLE
Why does an alert rule take effect on all business groups?

### DISCUSSION THREAD

**Discussion by czxin788** [ccfos/nightingale#1972] [Q&A] [answered]
我对某个业务组设置了告警规则，期望的是告警规则仅对当前业务组生效。
可实际效果是，我发现其他业务组的机器也匹配了当前业务组的告警规则。我感觉这么设计不合理。
怎么才能实现告警规则仅对当前业务组生效，不要对所有业务组都生效。
不知道是不是我使用方法有问题。
我用的版本是：v7.0.0-beta.5
![image](https://github.com/ccfos/nightingale/assets/13861904/d07121f5-04b8-48a2-9a42-3ae9b18062f7)

我们用的夜莺v2.6版本的告警规则，就是只对当前业务组生效，而不会对所有的业务组的机器生效，我感觉这个合理。
![image](https://github.com/ccfos/nightingale/assets/13861904/53b24641-e2e3-4711-8926-118fd93d4416)

**Comment by UlricQin** [comment, marked as answer]
这个说来话长。v7 和 v2 相比，告警机制上完全不同。v2 是自有体系，v7 是拥抱 prometheus 生态。v7 的 告警和恢复机制，这篇文章有讲：https://flashcat.cloud/docs/content/flashcat-monitor/nightingale-v6/faq/alerting-logic/ 如果你只是机器监控，告警规则里有个配置，叫仅在本业务组生效，能解决一部分常规问题

**Comment by czxin788** [comment]
好的，谢谢回复。
那我每次手动更新成仅在本业务组生效吧。

文章我看了，确实解决我的一些疑问。用高版本夜莺，曾经遇到过好多次故障其实没恢复，却莫名奇妙的报恢复了，过一会又报故障，原来是夜莺用promql没拿到值导致的。

以前用zabbix，故障恢复时会告诉恢复时的值和故障持续时长，通过对这篇文章的解读，也可以看出夜莺从底层原理来说，是无法实现这个功能的。

**Reply by UlricQin** [reply]
故障持续时长基本是可以拿到的：

```
{{$time_duration := sub now.Unix .FirstTriggerTime }}{{if .IsRecovered}}{{$time_duration = sub .LastEvalTime .FirstTriggerTime }}{{end}}**持续时长**: {{humanizeDurationInterface $time_duration}} 
```

新版本的夜莺的企微的 go template 里有上面的计算逻辑，可以作为持续时长。另外当前值其实也是可以的，只是会比较麻烦，近期会更新一个版本，把当前值的获取逻辑做得更简单。
